Dockerfile

```diff
@@ -26,26 +29,35 @@ RUN apt-get update \
     sudo \
     uuid-runtime \
     tzdata \
+  && apt-get clean \
   && rm -rf /var/lib/apt/lists/* \
   && localedef -i en_US -f UTF-8 en_US.UTF-8 \
   && useradd -m -s /bin/bash linuxbrew \
-  && echo 'linuxbrew ALL=(ALL) NOPASSWD:ALL' >>/etc/sudoers
+  && echo 'linuxbrew ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers \
+  && mkdir -p \
+     /home/linuxbrew/.linuxbrew/bin \
+     /home/linuxbrew/.linuxbrew/etc \
+     /home/linuxbrew/.linuxbrew/include \
+     /home/linuxbrew/.linuxbrew/lib \
+     /home/linuxbrew/.linuxbrew/opt \
+     /home/linuxbrew/.linuxbrew/sbin \
+     /home/linuxbrew/.linuxbrew/share \
+     /home/linuxbrew/.linuxbrew/var/homebrew/linked \
+     /home/linuxbrew/.linuxbrew/Cellar \
+  && ln -s ../Homebrew/bin/brew /home/linuxbrew/.linuxbrew/bin/ \
+  && mv /tmp/Homebrew /home/linuxbrew/.linuxbrew/Homebrew \
+  && chown -R linuxbrew: /home/linuxbrew/.linuxbrew \
+  && chmod -R g+w,o-w /home/linuxbrew/.linuxbrew
 
-COPY . /home/linuxbrew/.linuxbrew/Homebrew
-ENV PATH=/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:$PATH
+USER linuxbrew
 WORKDIR /home/linuxbrew
 
-# hadolint ignore=DL3003
-RUN cd /home/linuxbrew/.linuxbrew \
-  && mkdir -p bin etc include lib opt sbin share var/homebrew/linked Cellar \
-  && ln -s ../Homebrew/bin/brew /home/linuxbrew/.linuxbrew/bin/ \
-  && git -C /home/linuxbrew/.linuxbrew/Homebrew remote set-url origin https://github.com/Homebrew/brew \
-  && git -C /home/linuxbrew/.linuxbrew/Homebrew fetch origin \
+RUN \
+  git -C ~/.linuxbrew/Homebrew remote set-url origin https://github.com/Homebrew/brew \
+  && git -C ~/.linuxbrew/Homebrew fetch origin \
   && HOMEBREW_NO_ANALYTICS=1 HOMEBREW_NO_AUTO_UPDATE=1 brew tap homebrew/core \
   && brew install-bundler-gems \
   && brew cleanup \
-  && { git -C /home/linuxbrew/.linuxbrew/Homebrew config --unset gc.auto; true; } \
-  && { git -C /home/linuxbrew/.linuxbrew/Homebrew config --unset homebrew.devcmdrun; true; } \
-  && rm -rf ~/.cache \
-  && chown -R linuxbrew: /home/linuxbrew/.linuxbrew \
-  && chmod -R g+w,o-w /home/linuxbrew/.linuxbrew
```

Comment:
Are these definitely no longer needed i.e. `brew doctor` is 👍🏻 with these?

Reply:
They were not removed -- they were relocated to the first `RUN` instruction such that all root-user logic and actions are performed there. Check lines 49 and 50 in the modified version of this file.

Reply:
Ah, sorry, gotcha, thanks!